packages/snfoundry/scripts-ts/deploy-contract.ts

```diff
@@ -146,4 +157,39 @@ const exportDeployments = () => {
   fs.writeFileSync(networkPath, JSON.stringify(deployments, null, 2));
 };
 
+const deployScript = async (): Promise<void> => {
+  const walletAddress = deployer.address;
+  const isWalletDeployed = await checkWalletDeployed(walletAddress);
+
+  if (!isWalletDeployed) {
+    console.error(
+      "StarkNet wallet is not deployed. Please initialize and deploy your wallet before proceeding."
+    );
+    console.log("To deploy your wallet, follow these steps:");
+    console.log("1. Initialize your StarkNet wallet.");
+    console.log("2. Deploy the initialized wallet.");
+    console.log("3. Retry the contract deployment.");
+    return;
+  }
+
+  try {
```

Comment:
this is the core logic of deployment, we shouldnt deploy an specific contract here. deployments happen on `deploy.ts`